### PR TITLE
Add tests for basic UNPACK command behavior

### PR DESCRIPTION
**Pull request by jpd002**
Here's a test that covers some of UNPACK's behavior. There's lots of stuff that could be tested here, but I wanted to get some basic cases covered first. 

The `testSetUnpack` case is one case I encountered while doing some debugging in Kingdom Hearts 2. I thought it was interesting because it can be used to clear 4096 bytes in VU memory without supplying any data. (it was also broken in Play!)

### CODE REVIEW

File: tests/dma/vif/unpack.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	vifcode.NOP();
+	vifcode.STCYCL(4, 4);
+	vifcode.UNPACK(VIF::UNPACK_TYPE_S8, num, 0);
+	
+	DMA::SendSimple(unit->dmaChannel, vifcode.Raw(), packetSize);
+	
+	u32 vpsAfter = unit->regs->stat & VIF::STAT_VPS_MASK;
+	
+	printf("No data UNPACK(NUM: %d) -> VPS before: %d, VPS after: %d, CODE: %08x, NUM: %d\n", 
+	    num, vpsBefore, vpsAfter, unit->regs->code, unit->regs->num);
```
Comment (unknownbrackets) [inline tests/dma/vif/unpack.cpp:82]: Does this actually set the memory to anything?  Or just adjust the other regs?

-[Unknown]

**Review by unknownbrackets — APPROVED** [review]
That's a cool way to fill it - nice test.

Yeah, UNPACK seems like a pretty tricky workhorse.  I'm sure there will be more odd usage to test too.

-[Unknown]

File: tests/dma/vif/unpack.cpp (hunk 1 above)
Comment (jpd002) [reply tests/dma/vif/unpack.cpp:82]: Nope, I just wanted to have a look at the VIF's state when it's waiting for data to complete the UNPACK command.

Comment (unknownbrackets) [reply tests/dma/vif/unpack.cpp:82]: Ah, got it.

-[Unknown]

**Comment by jpd002** [comment]
Thanks for merging. Yeah, surely, there will be other tricky cases to cover. There's that Snowblind engine case that I'd like to tackle at some point because I'd like to get those games to work properly without having to activate the hack I've got in Play!.

**Pull request merged**